deluge/ui/hostlist.py

```diff
@@ -216,30 +216,42 @@ def on_connect_failed(reason, host_id):
             return defer.succeed(status_offline)
 
         try:
-            ip = gethostbyname(host)
+            # getaddrinfo returns a list of 5-tuples save this to list
+            addrinfo = getaddrinfo(host, None)
+            ip = []
+            for i in range(0, len(addrinfo)):
+                ip.append(addrinfo[i][4][0])
```

Comment:
change the `i in range` to `i in addrinfo` (and change the names to something even better)

Reply:
Why not something like:
```python
ips = [addrinfo[4][0] for addrinfo in getaddrinfo(host, None)]
```

Reply:
@aresch yeah, even better

Reply:
@aresch I guess that is possible, my python skills are very basic so, I often miss things like this. If that works I'll update the PR.